nautobot/dcim/elevations.py

```diff
@@ -217,7 +223,77 @@ def merge_elevations(self, face):
 
         return elevation
 
-    def render(self, face, unit_width, unit_height, legend_width):
+    def render_csv(self, face: Literal["front", "rear"]) -> str:
+        @dataclass
+        class DeviceCSVRepresentation:
+            name: Optional[int]
+            unit: str
+            device: "Device"
+
+        def _get_face_data(face: Literal["front", "rear"]) -> list[DeviceCSVRepresentation]:
+            # Prepare an empty rack representation
+            rack_positions = [None for x in range(0, self.rack.u_height)]
+
+            # Fill with used rack units
+            for unit in self.rack.get_rack_units(face=face, expand_devices=True):
+                if unit["device"] is None:
+                    rack_positions[unit["id"] - 1] = DeviceCSVRepresentation(name="", unit=unit["name"], device=None)
+
+                else:
+                    rack_positions[unit["id"] - 1] = DeviceCSVRepresentation(
+                        name=unit["device"].name,
+                        unit=unit["name"],
+                        device=unit["device"],
+                    )
+
+            return rack_positions
+
+        def _output_face_to_csv(
+            buffer: io.StringIO, rack_positions: list[DeviceCSVRepresentation], ru_prefix: str = ""
+        ):
+            if self.rack.desc_units:
+                position_iterator = range(self.rack.u_height)
+            else:
+                position_iterator = range(self.rack.u_height - 1, 0, -1)
+
+            for position in position_iterator:
+                this_position = rack_positions[position]
```

Comment:
If `rack_positions` were already ordered by `self.rack.desc_units`, this simplifies, right? To something like:

```suggestion
            for this_position in rack_positions:
```

Reply:
I see no mention of rack positions being outputted depending on desc_units from .get_rack_units(), maybe we should add that as well?